controllers/dscinitialization/auth.go

```diff
@@ -3,34 +3,74 @@ package dscinitialization
 import (
 	"context"
 
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	k8serr "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
 	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/services/v1alpha1"
 	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/components/dashboard"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
 )
 
-func (r *DSCInitializationReconciler) createAuth(ctx context.Context) error {
-	// Create Auth CR singleton
-	defaultAuth := client.Object(&serviceApi.Auth{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       serviceApi.AuthKind,
-			APIVersion: serviceApi.GroupVersion.String(),
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: serviceApi.AuthInstanceName,
-		},
-		Spec: serviceApi.AuthSpec{
-			AdminGroups:   []string{dashboard.GetAdminGroup()},
-			AllowedGroups: []string{"system:authenticated"},
-		},
-	},
-	)
-	err := r.Create(ctx, defaultAuth)
+func (r *DSCInitializationReconciler) createAuth(ctx context.Context, dscInit *dsciv1.DSCInitialization) error {
+	log := logf.FromContext(ctx)
+	// check for the dashboardConfig kind.
+	// Once the groupsConfig entry in the dashboardConfig is removed this logic can be removed.
+	crd := &apiextv1.CustomResourceDefinition{}
+	err := r.Client.Get(ctx, client.ObjectKey{Name: "odhdashboardconfigs.opendatahub.io"}, crd)
+
+	// can't find the dashboard kind so create the default.
+	if err != nil {
```

Comment:
In general, I would recommend to use `IsNotFound(err)` to check if the resource is actually not found, otherwise another error could be returned and not properly handled, i.e. a permission error would be handled the same as a NotFound one.

Something like:


```go
if err != nil && k8serr.IsNotFound(err) {
    // create
} else if err != nil {
    return fmt.Errorf("failed to get odhdashboardconfigs.opendatahub.io CRD: %w", err)
}
```

Reply:
:+1: